src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -733,27 +734,27 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
      *  runtime reflection universe, and if yes and if we've not yet loaded the requested info, then to commence initialization.
      */
     final def getFlag(mask: Long): Long = {
-      if (!isCompilerUniverse && !isThreadsafe(purpose = FlagOps(mask))) initialize
-      flags & mask
+      var f = if ((mask & PhaseIndependentFlags) == mask) rawflags else flags
+      if (((f & InCompilerUniverse) == 0) && !isThreadsafe(purpose = FlagOps(mask))) {
+        initialize
+        f = flags
+      }
+      f & mask
     }
```

Comment:
An alternative design would be to make `getFlag` non final, and override it in [`SynchronizedSymbol`](https://github.com/scala/scala/blob/e0967443a8fff62bb8a8ae193b4cb3f541a86fe4/src/reflect/scala/reflect/runtime/SynchronizedSymbols.scala#L34). Assuming that runtime reflection is not used in same classloader that is running the batch compiler, the JVM would treat this method as effectively final and inline it if needed.

We didn't have great performance testing infrastructure when we tried to accommodate the needs of runtime reflection into the compiler implementation, but now that we have better benchmarks, we can revisit these sort of implementation choices.